solr/core/src/java/org/apache/solr/api/Api.java

```diff
@@ -44,9 +43,7 @@ public Map<String, JsonSchemaValidator> getCommandSchema() {
         if (commandSchema == null) {
           ValidatingJsonMap commands = getSpec().getMap("commands", null);
           commandSchema =
-              commands != null
-                  ? ImmutableMap.copyOf(ApiBag.getParsedSchema(commands))
-                  : ImmutableMap.of();
+              commands != null ? Map.copyOf(ApiBag.getParsedSchema(commands)) : Map.of();
```

Comment:
Should we consider using `Collections.unmodifiableMap()` to get closer behavior to ImmutableMap? I don't think `Collection.unmodifiableMap` makes all of the same guarantees, but might make more clear that it is read only.

This seems to be true for a number of places in these changes where we go from an ImmutableCollection to just a regular Map or Set, wondering if that was intentional or if we should try to consistently keep those unmodifiable.

Reply:
This also goes for List.of and Set.of - the only case I found so far is that ImmutableSet from Guava cared about the order and that was an issue in one case. All other cases there was no difference that I found.